File: 2444-count-subarrays-with-fixed-bounds/2444-count-subarrays-with-fixed-bounds.kt
class Solution {
    fun countSubarrays(nums: IntArray, minK: Int, maxK: Int): Long {
        var answer = 0L
        var lastIndex = -1
        var minIndex = -1
        var maxIndex = -1
        for (i in nums.indices) {
            val num = nums[i]
            if (num !in minK..maxK) {
                lastIndex = i
            }

            if (num == minK) {
                minIndex = i
            }
            
            if (num == maxK) {
                maxIndex = i
            }

            if (lastIndex >= minIndex || lastIndex >= maxIndex) {
                continue
            }

            answer += min(minIndex, maxIndex) - lastIndex
        }

        return answer
    }
}

File: 2444-count-subarrays-with-fixed-bounds/2444-count-subarrays-with-fixed-bounds.py
class Solution:
    def countSubarrays(self, nums: List[int], minK: int, maxK: int) -> int:
        lastInvalidIndex = -1
        lastMinIndex = -1
        lastMaxIndex = -1
        answer = 0
        for index, num in enumerate(nums):
            if not (minK <= num <= maxK): lastInvalidIndex = index
            if num == minK: lastMinIndex = index
            if num == maxK: lastMaxIndex = index

            minIndex = lastMinIndex if lastMinIndex < lastMaxIndex else lastMaxIndex
            diff = minIndex - lastInvalidIndex
            if diff > 0: answer += diff

        return answer
        